### DISCUSSION TITLE
Convert local .mp4 files to HLS/DASH and upload to own CDN?

### DISCUSSION THREAD

**Discussion by mattirck** [HISPlayer/Unity_Video_Player#6] [Q&A]
Hi,

I just found your Asset and want to evaluate it for a possible use in company context. I really like the possibility to convert the videos into HLS and DASH streams for multi-platform delivery.

Is it with the SDK also possible to convert local .mp4-files to HLS only without uploading it to your portal?
We are using Azure Blobstorage to store our files and would like to upload them there.

**Comment by NurikHISPlayer** [comment]
Hi @mattirck ,

Unfortunately, that is not an option at the moment, but we will keep it in mind for our roadmap. 
On the other hand, our video player asset is fully compatible with Azure streams, no need to use our server.

**Reply by mattirck** [reply]
Thanks for the very fast answer, I really appreciate that.

I have another question: is it somehow possible to sent a Bearer Token for authorization as Custom HTTP Header Parameter?

Please see this screenshot:

![grafik](https://github.com/HISPlayer/Unity_Video_Player/assets/118195068/a7577032-884b-4db8-aa6c-a6c10146fe05)

This is a service-backend in Azure with anonymous access for testing purposes which delivers back a simple video-stream from a .mp4-file stored in a Blob storage which is working fine with your free trial.
But to access our productive service-backend a Bearer Token must be sent in the header to the endpoint-call because anonymous access is forbidden in enterprise context.

Is it possible to sent a Custom HTTP Header containing that token for authentication on all platforms? At the moment we would need Windows, UWP and Android. Maybe Apple Vision Pro later on but now right now.

**Reply by NurikHISPlayer** [reply]
This is not supported in the current version, we have added it to our roadmap and will add it, stay tuned.

**Reply by mattirck** [reply]
That would be really cool and a unique selling point as I had evaluated a lot of videoplayer assets in the past and near none had implemented it or just for HLS/DASH-streams and not for simple .mp4-files.

**Reply by mattirck** [reply]
> asset is fully compatible with Azure streams, no need to use our server.

True but it would be cool to have a feature included that could convert .mp4-files to HLS-format by code and let the developer do what ever he wants with it after that :)

**Reply by NurikHISPlayer** [reply]
Thanks for your very valuable feedback

**Reply by NurikHISPlayer** [reply]
Hi @mattirck,
Can we get a test content of Custom HTTP Headers for .mp4 file so that we can verify our implementation? 
You can send it to contact@hisplayer.com

**Reply by mattirck** [reply]
I really appreciate the speed you have developed this. But unfortunetely I can't give you access to our development-environment as this is accessible only from inside our company and you would need an employee-account, sorry.

Maybe it's an option for you to request a free Microsoft developer account and setup your own azure web app with restricted access that requires authentication? Then you can send a Bearer Token as authorization-header.

**Reply by NurikHISPlayer** [reply]
Thanks @mattirck , we will try it.
